core/src/repair_service.rs

```diff
@@ -518,20 +527,41 @@ impl RepairService {
         slot: Slot,
         slot_meta: &SlotMeta,
         max_repairs: usize,
+        delay_threshold: Duration,
     ) -> Vec<ShredRepairType> {
         if max_repairs == 0 || slot_meta.is_full() {
             vec![]
         } else if slot_meta.consumed == slot_meta.received {
+            // check delay time of last shred
+            if slot_meta.received > 0 {
+                if let Some(shred_data) = blockstore
+                    .get_data_shred(slot, slot_meta.received - 1)
+                    .unwrap()
+                {
+                    let reference_tick =
+                        u64::from(shred::layout::get_reference_tick(&shred_data).unwrap());
+                    let ticks_since_first_insert = DEFAULT_TICKS_PER_SECOND
+                        * (timestamp() - slot_meta.first_shred_timestamp)
```

Comment:
what if `timestamp() < slot_meta.first_shred_timestamp`?
`SystemTime` is not monotonic.

also can we avoid the `sleep` in the tests by adding a `now: u64` argument as the timestamp?

Reply:
Sure, I'm happy to add a time parameter and remove sleeps from code.

Reply:
changed to `saturating_sub` to account for non-monotonic timestamps.